mne/decoding/transformer.py

```diff
@@ -60,29 +77,35 @@ def fit(self, epochs_data, y):
         self : instance of Scaler
             Returns the modified instance.
         """
+        from sklearn.base import clone
         if not isinstance(epochs_data, np.ndarray):
             raise ValueError("epochs_data should be of type ndarray (got %s)."
                              % type(epochs_data))
 
         X = np.atleast_3d(epochs_data)
+        n_epochs, n_channels, n_times = X.shape
+        X = X.transpose(1, 0, 2)
 
-        picks_list = dict()
-        picks_list['mag'] = pick_types(self.info, meg='mag', ref_meg=False,
-                                       exclude='bads')
-        picks_list['grad'] = pick_types(self.info, meg='grad', ref_meg=False,
-                                        exclude='bads')
-        picks_list['eeg'] = pick_types(self.info, eeg=True, ref_meg=False,
-                                       meg=False, exclude='bads')
+        if self.scaler == 'standard':
+            for (_, pick) in self._pick_types:
+                if self.with_mean:
+                    ch_mean = X[pick].mean(axis=1)[:, None, :]
+                    self.ch_mean_.append(ch_mean)  # TODO rename attribute
+            if self.with_std:
+                ch_std = X[pick].mean(axis=1)[:, None, :]
+                self.std_.append(ch_std)  # TODO rename attribute
 
-        self.picks_list_ = picks_list
+        elif not isinstance(self.scaler, str):
+            X = X.reshape(n_channels, -1)
+            self.scalers_ = list()
+            for (_, pick) in self._pick_types:
+                scaler = clone(self.scaler)
+                scaler.fit(X[pick].T)
+                self.scalers_.append(scaler)
 
-        for key, this_pick in picks_list.items():
-            if self.with_mean:
-                ch_mean = X[:, this_pick, :].mean(axis=1)[:, None, :]
-                self.ch_mean_[key] = ch_mean  # TODO rename attribute
-            if self.with_std:
-                ch_std = X[:, this_pick, :].mean(axis=1)[:, None, :]
-                self.std_[key] = ch_std  # TODO rename attribute
+        elif self.scaler == 'default':
+            # we will directly transform by calling function
+            pass
```

Comment:
this is problematic ... you cannot do this. Scaling should always be estimated on training data